web/main.js

```diff
@@ -1,164 +1,261 @@
-  metadata = await getData("metadata.json");
-  reviews = await getData("reviews.json");
-  discussions = await getData("discussions.json");
-};
-
-const render = () => {
-  const ctnr = document.querySelector("#container");
-  ctnr.innerHTML = "";
-
-  renderHeader(ctnr);
-  renderDiscussions(ctnr);
-  renderReviews(ctnr);
-};
-
-// Magically replaces all template values with the value from the dictionary.
-// e.g. replaces {game} with the value of data["game"]
-const applyDataToTemplate = (data, template) =>
-  template.replaceAll(/\{([^\}]+)\}/g, (_, key) => data[key]);
-
-const getTime = (daysAgo) => 
-  daysAgo >= 365 ? Math.floor(daysAgo / 365) + " years" : daysAgo + " days";
-
-const renderHeader = (ctnr) => {
-  { // Container
-    const template = document.querySelector("#header-container-template").innerHTML;
-    const html = template;
-    ctnr.insertAdjacentHTML("beforeend", html);
+      setLocalStorageJSONObject("readReviews", readReviews);
+    }
   }
 
-  {
-    // summary of individual records per-game
-    
-    // amend metadata to add number of reviews to each game
-    const reviewsPerGame = {};
+  window.markDiscussionAsRead = (event) => {
+    const star = event.currentTarget.parentNode.querySelector(".unread-star");
+    const key = star.getAttribute("data-key");
+    readDiscussions[key] = true;
+    setLocalStorageJSONObject("readDiscussions", readDiscussions);
+
+    star.classList.add("hidden");
+  }
+
+  window.markReviewAsRead = (event) => {
+    const star = event.currentTarget.parentNode.querySelector(".unread-star");
+    const key = star.getAttribute("data-key");
+    readReviews[key] = true;
+    setLocalStorageJSONObject("readReviews", readReviews);
+
+    star.classList.add("hidden");
+  }
+
+  const getLocalStorageJSONObject = (key) => {
+    const jsonStr = localStorage.getItem(key);
+    if (jsonStr === null)
+      return null;
 
-    Object.entries(reviews).forEach(review =>
+    try
     {
-      var reviewData = review[1];
-      var appId = reviewData.app_id;
-      if (!(appId in reviewsPerGame))
-      {
-        reviewsPerGame[appId] = 0;
-      }
+      return JSON.parse(jsonStr);
+    }
+    catch
+    {
+      localStorage.removeItem(key);
+      return null;
+    }
+  }
 
-      reviewsPerGame[appId]++;
+  const setLocalStorageJSONObject = (key, obj) =>
+    localStorage.setItem(key, JSON.stringify(obj));
+
+  const getData = async (file) => {
+    var headers = new Headers();
+    headers.append("pragma", "no-cache");
+    headers.append("cache-control", "no-cache");
+
+    const res = await fetch(`data/${file}`, {
+      method: "GET",
+      headers: headers
     });
 
-    // amend metadata to add number of discussions to each game
-    const discussionsPerGame = {};
+    var toReturn = await res.json();
+    return toReturn;
+  };
+
+  const populateState = async () => {
+    readReviews = getLocalStorageJSONObject("readReviews");
+    if (readReviews === null)
```

Comment:
I prefer curly-braces always; it's easy to not notice them and introduce bugs by adding other statements after this.

Reply:
Added curly-braces in aadf315411b2162df5332430ecf89e1a5a9f61e9